mlos_core/mlos_core/spaces/adapters/adapter.py

```diff
@@ -44,28 +44,28 @@ def target_parameter_space(self) -> ConfigSpace.ConfigurationSpace:
         pass  # pylint: disable=unnecessary-pass # pragma: no cover
 
     @abstractmethod
-    def transform(self, configuration: pd.DataFrame) -> pd.DataFrame:
+    def transform(self, configuration: pd.Series) -> pd.Series:
```

Comment:
Trying to decide if we should still have a bulk version of this or use it as an `apply` over a DataFrame's rows.

Reply:
This part might have actually been better split to a separate PR.  I'm not sure how necessary it is now, and maybe we leave it, but it could have made some of the other changes simplier to leave them as DataFrames.